swap_meet/vendor.py

```diff
@@ -1,2 +1,59 @@
 class Vendor:
-    pass
+    def __init__(self, inventory=None):
+        self.inventory = inventory if inventory is not None else []
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+
+        return False
+
+    def get_by_category(self, category):
+        item_in_category = []
+        for item in self.inventory:
+            if item.category == category:
+                item_in_category.append(item)
+        
+        return item_in_category
+
+    def swap_items(self, vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in vendor.inventory:
+            return False
```

Comment:
Great guard clause!

Notice that here, the explicit check is superior to trying to remove an item and handling the `ValueError` that might happen. We might run into a situation where the first remove succeeds, but the second fails. This would require us to write code to put things back the way they had been. It's cleaner to check whether everything is in a valid state to begin with.